packages/hub/src/app/models/[owner]/[slug]/revisions/[revisionId]/ModelRevisionView.tsx

```diff
@@ -59,9 +62,13 @@ export const ModelRevisionView: FC<{
 
   const typename = model.revision.content.__typename;
   if (typename !== "SquiggleSnippet") {
-    return <div>Unknown model type {typename}</div>;
+    throw new Error(`Unknown model type ${typename}`);
```

Comment:
why the change? I'm nervous about ``throw`` in the ``hub`` components.

Reply:
Because I couldn't do an early return. There's `useValidSquiggleVersion` hook call below this line, which relies on `typename` being `"SquiggleSnippet"`.

This error should never be triggered and it's mostly for TypeScript. We don't have other typenames (model types) yet, and when we will add them, we're going to review everything and change this, split this component and extract model-type-specific parts each in a separate component, similar to how we do this with `EditModelPage` -> `EditSquiggleSnippetModel`.

Reply:
Maybe we could add ``UnreachableError`` or something, at some point, to make it clearer that this is intended to be unreachable and is mainly for TS.

Reply:
It's possible, but for it to be useful, we'd need to check for it in `ErrorBoundary` code, and I'm not sure how that would look like.

Another thing we could try is an exhaustiveness check. `typename` is initially `"%other" | "SquiggleSnippet"`. I'd have to do more experiments to understand if `"%other"` is because `SquiggleSnippet` is the only fragment that we asked in the query, or if it's possible for relay-compiler to list all possible typenames because it read our graphql schema. If relay-generated types would be exhaustive then we could do a switch statement with `satisfies never` trick at the end.